Add support for several Qt flavors

Qt has two flavors: X11 and embedded (QWS). QWS has some limitations
compared to X11. This patch adds an option to select which version to
use.

Signed-off-by: Gwenhael Goavec-Merou <gwenhael.goavec-merou@trabucayre.com>
Signed-off-by: Akihiko Odaki <akihiko.odaki.4i@stu.hosei.ac.jp>
---
Index: PyQt5_gpl-5.6/configure.py
===================================================================
--- PyQt5_gpl-5.6.orig/configure.py
+++ PyQt5_gpl-5.6/configure.py
@@ -959,6 +959,9 @@
         if opts.vendlibdir is not None:
             self.vend_lib_dir = opts.vendlibdir
 
+        if opts.qtflavor is not None:
+            self.qt_flavor = opts.qtflavor
+
         # Handle any conflicts.
         if not self.qt_shared:
             if not self.static:
@@ -1195,6 +1198,11 @@
             dest="notools",
             help="disable the building of pyuic5, pyrcc5 and pylupdate5 "
                     "[default: enabled]")
+    g.add_option("--qt-flavor", dest='qtflavor', type='string',
+            default="WS_X11", metavar='QTFLAVOR', action='store',
+            help="select Qt flavor beetween WS_X11 and WS_QWS "
+                    "only available on Linux)"
+                    "[default: WS_X11]")
     p.add_option_group(g)
 
     # Vendor ID.
@@ -2280,7 +2288,7 @@
     elif target_config.py_platform == 'darwin':
         plattag = 'WS_MACX'
     else:
-        plattag = 'WS_X11'
+        plattag = target_config.qt_flavor
 
     sip_flags.append('-t')
     sip_flags.append(plattag)
